### PR TITLE
docs: Update details about compatible existing GPU Pods

### PR DESCRIPTION
**Pull request by eahydra**
Signed-off-by: Joseph <joseph.t.lee@outlook.com>

### Ⅰ. Describe what this PR does

Update details about compatible existing GPU Pods.

### V. Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `make test`

### CODE REVIEW

File: docs/proposals/scheduling/20220629-fine-grained-device-scheduling.md [hunk 1]
```diff
@@ -355,6 +341,59 @@ type DeviceList struct {
 }
 ```
 
+##### Compatible
+
+Considering that some users already have many existing GPU Pods in their clusters, it is necessary to ensure that Koordinator GPU Scheduling does not repeatedly allocate the GPU devices held by these GPU Pods. Therefore, koord-scheduler needs to obtain the GPU devices's information held by these existing Pods. These GPU devices are allocated by the kubelet and recorded in the local file `/var/lib/kubelet/device-plugins/kubelet_internal_checkpoint`, so the device reporter will parse the file to obtain the GPU Device ID assigned to each Pod. When parsing, it needs to exclude the Pod that allocates GPU through koord-scheduler, and finally update it to Device CRD in the form of annotation. The corresponding annotation key is `node.koordinator.sh/gpu-checkpoints`, and the annotation value is defined as follows:
```
Comment (WeizhouRen) [inline docs/proposals/scheduling/20220629-fine-grained-device-scheduling.md:346]: Don't you need to patch the allocation result into existing Pods?

File: docs/proposals/scheduling/20220629-fine-grained-device-scheduling.md [hunk 2]
```diff
@@ -355,6 +341,59 @@ type DeviceList struct {
...
+type PodDevicesEntries []PodDevicesEntry
+```
+
+#### CRD Example
+```yaml
+apiVersion: scheduling.koordinator.sh/v1alpha1
+kind: Device
+metadata:
+  name: node-1
+  annotations:
```
Comment (WeizhouRen) [inline docs/proposals/scheduling/20220629-fine-grained-device-scheduling.md:366]: What's the point of updating the annotation value in Device CRD?

File: docs/proposals/scheduling/20220629-fine-grained-device-scheduling.md (hunk 1 above)
Comment (eahydra) [reply docs/proposals/scheduling/20220629-fine-grained-device-scheduling.md:346]: Restrict koordlet permissions. Koordlet does not require Pod write permissions, and in some user scenarios, permissions on a single machine are also restricted.

File: docs/proposals/scheduling/20220629-fine-grained-device-scheduling.md [hunk 3]
```diff
@@ -382,27 +422,38 @@ type ContainerResourceHookResponse struct {
...
+3. Wait a few minutes to ensure that each node's koordlet creates and updates the Device CRD.
+4. Modify all components that create GPU Pods to switch the schedulerName of the Pod to koord-scheduler
+5. Start trying to create a GPU Pod and verify the koord-scheduler GPU Scheduling scheduling result.
+6. Restore the components that created the GPU Pod and the old scheduler.
 
-If there is no GPU device plugin, the only different is in pre-bind stage, we will erase the origin-gpu-resource-dimension
-in container to pass kubelet resource limitation check. This will be a flag in scheduler side.
+In the future Koordinator will provide a webhook to solve the upgrade problem. The webhook will identify the GPU Pod and modify the schedulerName of the newly created GPU Pod to koord-scheduler. At the same time, the webhook will take over the Binding operation of the GPU Pod. If the Binding is not initiated by koord-scheduler, it will be rejected.
 
 ## Unsolved Problems
```
Comment (jasonliu747) [inline docs/proposals/scheduling/20220629-fine-grained-device-scheduling.md:458]: Webhook 相关信息加到 Unsolved Problems 这儿吧

File: docs/proposals/scheduling/20220629-fine-grained-device-scheduling.md [hunk 4]
```diff
@@ -382,27 +422,38 @@ type ContainerResourceHookResponse struct {
...
-container env, but we will overwrite these envs in runtimeproxy by allocation result in pod's annotation.
+stage, we will update container resources with Koordinator GPU resources, this is for kubelet to check resource limitation.
+We will also add device allocation information to Pod's annotation. In node side, the k8s device plugin will first patch
+container env, but we will overwrite these envs in runtimeproxy by allocation result in Pod's annotation.
+
+### Upgrade strategy
+
+If using Koordinator GPU Scheduling to schedule GPU Pods in a brand new cluster, simply install Koordinator components.
+
+However, if you want to upgrade to Koordinator GPU Scheduing in an existing cluster, you need to avoid GPU devices being repeatedly allocated because of switching between different scheduling mechanisms. You need to pay attention to the order when upgrading:
```
Comment (jasonliu747) [inline docs/proposals/scheduling/20220629-fine-grained-device-scheduling.md:447]: 最好也说一下为什么要按照这个顺序，原因是什么，如果顺序错误后果是什么

File: docs/proposals/scheduling/20220629-fine-grained-device-scheduling.md [hunk 5]
```diff
@@ -382,27 +422,38 @@ type ContainerResourceHookResponse struct {
...
+1. Install the Koordinator components. In particular, make sure that the koordlets are all started successfully.
+2. Stop the component that creates the new GPU Pod.
```
Comment (jasonliu747) [inline docs/proposals/scheduling/20220629-fine-grained-device-scheduling.md:449]: 你想表达的意思是“暂停申请带有GPU资源的Pod吗”？

Comment (eahydra) [reply docs/proposals/scheduling/20220629-fine-grained-device-scheduling.md]: 是的。这里应该说system or platform 可能更合适一些。已经修改。

**Review by jasonliu747 — APPROVED** [review]
/lgtm

**Review by buptcozy — APPROVED** [review]
/LGTM

**Pull request merged**